packages/filebrowser/src/listing.ts

```diff
@@ -2237,12 +2244,23 @@ export namespace DirListing {
         node.removeAttribute('data-is-dot');
       }
       // If an item is being edited currently, its text node is unavailable.
+      const indices = !model.indices ? [] : model.indices;
+      let highlightedName = StringExt.highlight(model.name, indices, h.mark);
       if (text) {
-        const indices = !model.indices ? [] : model.indices;
-        let highlightedName = StringExt.highlight(model.name, indices, h.mark);
         VirtualDOM.render(h.span(highlightedName), text);
       }
 
+      // Adds an aria-label to the checkbox element.
+      const checkboxInput = checkboxWrapper.querySelector(
+        'input[type="checkbox"]'
+      );
+      checkboxInput?.setAttribute(
+        'aria-label',
+        `Select ${
+          fileType.contentType === 'directory' ? 'directory' : 'file'
+        } '${highlightedName}'`
```

Comment:
Will need to use translations with gettext formatting for arguments.

Reply:
I don't know how and when to use `gettext`. Is that something like below ?
```
trans.__(trans.gettext('Select %1 \'$2\'', fileType.contentType === 'directory' ? 'directory' : 'file', highlightedName))
```

Reply:
I think that gettext does not need to be explicitly invoked, the following should work according to our documentation:

```ts
trans.__('Select %1 "%2"', fileType.contentType === 'directory' ? trans.__('directory') : trans.__('file'), highlightedName)
```

But I would actually recommend moving the branching logic out of translation string as we would not be able to translate nouns in context.

```ts
fileType.contentType === 'directory' ? trans.__('Select directory "%1"') : ? trans.__('Select file "%1"')
```

This is because "file" or "directory" may be translated to multiple words depending on context (e.g. here in Polish I would use "plik" but elsewehere it would mean "kartoteka"). Joining them together makes context collisions less likely and easier for translators to understand what is happening.

Reply:
Thanks a lot for the suggestions and details